src/coreclr/vm/dllimport.cpp

```diff
@@ -3316,8 +3316,11 @@ BOOL NDirect::MarshalingRequired(
                 FALLTHROUGH;
 
             case ELEMENT_TYPE_VALUETYPE:
+            case ELEMENT_TYPE_GENERICINST:
             {
                 TypeHandle hndArgType = arg.GetTypeHandleThrowing(pModule, &emptyTypeContext);
+                if(!hndArgType.IsValueType())
```

Comment:
Perhaps there is something above, but this doesn't seem fully sufficient. What about is a value type with a managed field?

Reply:
Ah. The logic is below for this case.